.github/workflows/release.yml

```
@@ -0,0 +1,50 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - v**
+
+jobs:
+  go-generator:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.20"
+          check-latest: true
+          cache: true
```

we can use the newest version of the action which has caching enabled by default:

```suggestion
      - name: Set up Go
        uses: actions/setup-go@v4
        with:
          go-version: "1.20"
          check-latest: true
```